src/public_key.rs

```diff
@@ -235,7 +235,19 @@ impl std::fmt::Display for PublicKey {
     }
 }
 
-use serde::de::{self, Deserialize, Deserializer, Visitor};
+use serde::{
+    de::{self, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+impl Serialize for PublicKey {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
```

Comment:
doesn't this assume that the only way o serialize is to a string format? Which is not what you'd use in, protos or in a wallet where you'd want the bytes form of this. How to reconcile those?

Reply:
You can't choose with serde, so you would have to use a separate serializing function. Strings were chosen here because it aligns with the current existing deserialize impl (which shares the same problem). 

An alternate choice would be to use `.serialize_bytes`, which would get what you want (would be done similarly for deserialize). 

Basically, the data types you get to choose from are those specified in the `Serializer` trait: https://docs.rs/serde/latest/serde/ser/trait.Serializer.html#required-methods

Reply:
If this matched Deserialize then the string approach is fine.. There's already a to_bytes function defined so we'd use that for the binary representation.